internal/routing/handlers/post_handler.go

```diff
@@ -109,7 +115,157 @@ func (handler *PostHandler) CreatePost(w http.ResponseWriter, r *http.Request) {
+	lastPostId := r.URL.Query().Get(utils.PostIdParamKey)
+	limit := r.URL.Query().Get(utils.LimitParamKey)
+
+	var userId interface{}
+	var countQuery string
+	var dataQuery string
+	var countQueryArgs []interface{}
+	var dataQueryArgs []interface{}
+
+	if private {
+		userId = claims.(jwt.MapClaims)["sub"]
+		countQuery = `SELECT COUNT(*) FROM alpha_schema.posts
+    					INNER JOIN alpha_schema.users ON author_id = user_id
+    					INNER JOIN alpha_schema.subscriptions ON user_id = subscriptions.subscriber_id
+    					WHERE subscriptions.subscriber_id = $1`
+		dataQuery = `SELECT post_id, username, nickname, profile_picture_url, content, posts.created_at FROM alpha_schema.posts
+					INNER JOIN alpha_schema.users ON author_id = user_id
+					INNER JOIN alpha_schema.subscriptions ON user_id = subscriptions.subscriber_id
+					WHERE subscriptions.subscriber_id = $1`
+
+		countQueryArgs = append(countQueryArgs, userId)
+		dataQueryArgs = append(dataQueryArgs, userId)
+	} else {
+		countQuery = "SELECT COUNT(*) FROM alpha_schema.posts"
+		dataQuery = `SELECT post_id, username, nickname, profile_picture_url, content, posts.created_at FROM alpha_schema.posts
+					INNER JOIN alpha_schema.users ON author_id = user_id`
+	}
+
+	// Get the count of posts in the database that match the criteria
+	row := tx.QueryRow(ctx, countQuery, countQueryArgs...)
+
+	var count int
+	if err := row.Scan(&count); err != nil {
+		utils.WriteAndLogError(w, schemas.DatabaseError, http.StatusInternalServerError, err)
+		return nil, 0, "", "", err
+	}
+
+	var rows pgx.Rows
+	var err error
+
+	if lastPostId == "" {
+		// If we don't have a last post ID, we'll return the newest posts
+		dataQuery += " ORDER BY created_at DESC LIMIT $2"
+		dataQueryArgs = append(dataQueryArgs, limit)
+	} else {
+		// If we have a last post ID, we'll return the posts that were created before the last post
+		dataQuery += " AND posts.created_at < (SELECT created_at FROM alpha_schema.posts WHERE post_id = $2) ORDER BY created_at DESC LIMIT $3"
+		dataQueryArgs = append(dataQueryArgs, lastPostId, limit)
+	}
+
+	rows, err = tx.Query(ctx, dataQuery, dataQueryArgs...)
+	if err != nil {
+		utils.WriteAndLogError(w, schemas.DatabaseError, http.StatusInternalServerError, err)
+		return nil, 0, "", "", err
+	}
+
+	// Iterate over the rows and create the post DTOs
+	intLimit, err := strconv.Atoi(limit)
+	if err != nil {
+		utils.WriteAndLogError(w, schemas.BadRequest, http.StatusBadRequest, err)
+		return nil, 0, "", "", err
+	}
+
+	posts := make([]*schemas.PostDTO, intLimit)
+
+	for rows.Next() {
+		post := &schemas.PostDTO{}
+		if err := rows.Scan(&post.PostId, &post.Author.Username, &post.Author.Nickname, &post.Author.ProfilePictureURL, &post.Content, &post.CreationDate); err != nil {
+			utils.WriteAndLogError(w, schemas.DatabaseError, http.StatusInternalServerError, err)
+			return nil, 0, "", "", err
+		}
+
+		posts = append(posts, post)
```

Comment:
Was passiert wenn das intLimit erreicht wird? Kommt dann ein error oder wird der for-Loop unterbrochen?

Reply:
Wie oben gesagt, wir limitieren in der Query bereits auf das `LIMIT`, weswegen wir den for-loop nicht unterbrechen müssen.